### ISSUE TITLE
Error code of  DeepONetCartesianProd in deeponet.py

### ISSUE THREAD

**Issue by Jianguo99** [lululxvi/deepxde#979]
when I try to reproduce the results of SDE's figure in deeponet's paper,I found that the line **62** of deeponet.py  is wrong, it will prouduce a matrix with size (batchsize,batchsize), but we need (batchsize,1).
I correct the code , as  following:
```python
 x = torch.sum(x_func * x_loc, dim=-1, keepdim=True)
```

Is right? Or a worng of my understanding for deeponet？

> It seems that the shape is (batchsize, n), not (batchsize, bachsize).

 x_func's and x_loc's shape are  (batchsize, n), but the following code will produce a matrix with shape (batchsize, bachsize) which is wrong.
```python
#the line 62 of deeponet.py
x = torch.einsum("bi,ni->bn", x_func, x_loc)
```

**Comment by OverKit** [comment]
It seems that the shape is (batchsize,  n), not  (batchsize, bachsize).

**Comment by jontyhuang-jg** [comment]
@OverKit  The y is a single point which is the definition of deepONet , so we need output shape `(batchsize,1)`. The code `torch.einsum("bi,ni->bn", x_func, x_loc)` should be transformed to  **element-wise multiplication**. I have try the code `x = torch.sum(x_func * x_loc, dim=-1, keepdim=True)`, it works and I get the same results with the paper's.

**Comment by lululxvi** [comment]
`x_func` has shape (batch size, ?) and `x_loc` has shape (n, ?). Have you also checked at `DeepONet` code?

**Comment by jontyhuang-jg** [comment]
> `x_func` has shape (batch size, ?) and `x_loc` has shape (n, ?). Have you also checked at `DeepONet` code?

I have checked DeepONetCartesianProd's codes. I set batchsize=1024 in deeponet_dataset.py ,then print the shapes:
![image](https://user-images.githubusercontent.com/80951029/199145054-b7826e63-7eba-4ed6-8ffe-aee7e4561d87.png)

**Comment by lululxvi** [comment]
In your case, `x_func` and `x_loc` has the same batch size (1024), but this is usually not the same. I am not sure if you use `DeepONetCartesianProd` in the expected way. Have you checked the two codes at https://deepxde.readthedocs.io/en/latest/demos/operator.html ?
